docs/coding-guidelines/vectorization-guidelines.md

```diff
@@ -0,0 +1,1091 @@
+- [Introduction to vectorization with Vector128 and Vector256](#introduction-to-vectorization-with-vector128-and-vector256)
+  * [Code structure](#code-structure)
+    + [Testing](#testing)
+    + [Benchmarking](#benchmarking)
+      - [Custom config](#custom-config)
+      - [Memory alignment](#memory-alignment)
+        * [Enforcing memory alignment](#enforcing-memory-alignment)
+        * [Memory randomization](#memory-randomization)
+  * [Loops](#loops)
+    + [Scalar remainder handling](#scalar-remainder-handling)
+    + [Vectorized remainder handling](#vectorized-remainder-handling)
+    + [Access violation testing](#access-violation-av-testing)
+  * [Loading and storing vectors](#loading-and-storing-vectors)
+    + [Loading](#loading)
+    + [Storing](#storing)
+    + [Casting](#casting)
+  * [Mindset](#mindset)
+    + [Edge cases](#edge-cases)
+    + [Scalar solution](#scalar-solution)
+    + [Vectorized solution](#vectorized-solution)
+  * [Toolchain](#toolchain)
+    + [Creation](#creation)
+    + [Bit operations](#bit-operations)
+    + [Equality](#equality)
+    + [Comparison](#comparison)
+    + [Math](#math)
+    + [Conversion](#conversion)
+    + [Widening and Narrowing](#widening-and-narrowing)
+    + [Shuffle](#shuffle)
+  * [Summary](#summary)
+    + [Best practices](#best-practices)
+
+TL;DR: Go to [Summary](#summary)
+
+# Introduction to vectorization with Vector128 and Vector256
```

Comment:
```suggestion
# Introduction to vectorization with Vector128 and Vector256
```
With `Vector64<T>`, `Vector128<T>`, `Vector256<T>`, and `Vector512<T>` getting introduced in this section, it would be helpful to also introduce `Vector`, `Vector<T>`, `Vector2`, `Vector3`, and `Vector4` and describe how their roles.

I recommend that @tannergooding adds that in as a follow-up PR though.

Reply:
👍. Notably `Vector2/3/4` are effectively specialized ones for graphics support and not for more general algorithms.